server/src/main/java/org/elasticsearch/rest/DeprecationRestHandler.java

```diff
@@ -65,9 +76,17 @@ public DeprecationRestHandler(
     @Override
     public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
         if (compatibleVersionWarning == false) {
-            deprecationLogger.warn(DeprecationCategory.API, deprecationKey, deprecationMessage);
+            if (deprecationLevel == null || deprecationLevel == Level.WARN) {
```

Comment:
i was going to comment about this having a default value. But I see that a level is different for compatible and different for "current version" deprecations.
might be worth a comment why deprecationLevel is nullable and why there is a different behaviour?